compiler/rustc_resolve/src/macros.rs

```diff
@@ -769,6 +769,15 @@ impl<'a> Resolver<'a> {
                     let expected = kind.descr_expected();
                     let msg = format!("cannot find {} `{}` in this scope", expected, ident);
                     let mut err = self.session.struct_span_err(ident.span, &msg);
+                    if ident.to_string() == "default"
```

Comment:
(nit: you could also do `ident.name == kw::default` to avoid the `to_string` by comparing just `Symbol`s)